File: Python/textbook_track/chapter03/ba3b.py
# Reconstruct a String from its Genome Path
import sys


def calc_string_spelled_by_a_genome_path(k_mers):
    text = list(k_mers[0])
    for k_mer in k_mers[1:]:
        text.append(k_mer[-1])
    return ''.join(text)


def main():
    data = sys.stdin.read().splitlines()
    result = calc_string_spelled_by_a_genome_path(data)
    print(result)


if __name__ == '__main__':
    main()

File: Julia/textbook_track/chapter03/ba3b.jl
# Reconstruct a String from its Genome Path

function calc_string_spelled_by_genome_path(kmers)
    text = collect(kmers[1])
    for kmer in kmers[2:end]
        push!(text, kmer[end])
    end
    join(text)
end


function main()
    data = readlines()
    result = calc_string_spelled_by_genome_path(data)
    println(result)
end


main()
